bn-apps/cordapp-updates-distribution/design/design.md

```diff
@@ -0,0 +1,135 @@
+![Corda](https://www.corda.net/wp-content/uploads/2016/11/fg005_corda_b.png)
+
+# CordApp distribution service (CDS)
+
+DOCUMENT MANAGEMENT
+---
+
+## Document Control
+
+| Title                | CordApp distribution service                             |
+| -------------------- | ------------------------------------------------------------ |
+| Date                 | 22 Aug 2018                                                |
+| Author               | Ivan Schasny, Mike Hearn |
+| Distribution         | Design Review Board, Product Management, Solutions Engineering, Platform Delivery |
+| Corda target version | OS                                                   |
+
+## HIGH LEVEL DESIGN
+
+### Overview
+
+This proposal describes the architecture of a reference implementation for a CorDapp Distribution Service (CDS), that would allow BNOs to distribute updates to CorDapps to Business Network participants. This design targets scenarios when Business Network members maintain their nodes by themselves and can decide on what version of a CorDapp they would like to upgrade to. This design does *not* target scenarios where upgrade function is centralised and the whole network can be shut down to perform stop-upgrade-restart.
+
+### Requirements
+
+The following requirements have been gathered from various internal discussions within R3 and from [groups.io mailing lists](https://groups.io/g/corda-dev/topic/22686107#190), which is recommended for reading prior to the design.
+
+* BNO should be able to notify members about new CorDapp version availability.
```

Comment:
Just seeing this design now.

Qq: When talking about "CorDapp", I assume this means both the contract code and the flows code, which live in separate jars, and will have a different lifecycle?

Reply:
Yeah, great point @tudor-malene . I guess the focus of this design is on _distribution_ of the apps. I.e. how to physically get jar files to an end-user's computer. That's why it explicitly says in the beginning that this design doesn't aim on automating _installation_ of the cordapps. Node admins would still have to stop / upgrade / restart etc.